packages/components/src/menu-item/style.scss

```diff
@@ -8,6 +8,7 @@
 			// Ensure unchecked items have clearance for consistency
 			// with checked items containing an icon or shortcut.
 			padding-right: $grid-unit-60;
+			box-sizing: initial;
```

Comment:
It looks like the recent changes to reset box-sizing on everything has negatively impacted the approach taken in https://github.com/WordPress/gutenberg/pull/34406 to ensure consistent menu item widths when handling both checked and unchecked items.

Switching box-sizing back in this circumstance smooths things out but it might be worth improving these styles in a follow-up.